src/texstudio.cpp

```diff
@@ -1057,9 +1057,45 @@ void Texstudio::setupMenus()
 	menu->addSeparator();
     newManagedAction(menu, "pasteAsLatex", tr("Pas&te as LaTeX"), SLOT(editPasteLatex()), QKeySequence(Qt::CTRL | Qt::SHIFT | Qt::Key_V), "editpaste");
 	newManagedAction(menu, "convertTo", tr("Co&nvert to LaTeX"), SLOT(convertToLatex()));
+
+	menu->addSeparator();
     newManagedAction(menu, "previewLatex", tr("Pre&view Selection/Parentheses"), SLOT(previewLatex()), Qt::ALT | Qt::Key_P);
 	newManagedAction(menu, "removePreviewLatex", tr("C&lear Inline Preview"), SLOT(clearPreview()));
 
+	submenu = newManagedMenu(menu, "previewMode", tr("Preview Dis&play Mode"));
+	previewModeGroup = new QActionGroup(this);
+	act = newManagedAction(submenu, "preview1", tr("Show preview as tooltip if panel is hidden"), SLOT(setPreviewMode()));
+	act->setData(ConfigManager::PM_TOOLTIP_AS_FALLBACK);
+	act->setCheckable(true);
+	previewModeGroup->addAction(act);
+	act = newManagedAction(submenu, "preview2", tr("Always show preview in preview panel"), SLOT(setPreviewMode()));
```

Comment:
menu names should be, at least a little, self-explanatory. 
preview2 (etc.) is definitely not. 
There is a reason why the enum is called PM_PANEL.